File: packages/shared/schemas_py/project.py
from pydantic import BaseModel
from typing import Optional, List

class Character(BaseModel):
    name: str
    description: str
    visual_description: str

class Scene(BaseModel):
    title: str
    description: str
    quote: Optional[str] = None
    mood: str
    visual_description: str
    scene_type: str = "introduction"

class BookAnalysis(BaseModel):
    summary: str
    themes: List[str]
    genre: str
    mood: str
    target_audience: str
    characters: List[Character]
    key_scenes: List[Scene]

class Project(BaseModel):
    id: str
    title: str
    author: str = ""
    description: str = ""
    book_file_url: Optional[str] = None
    cover_image_url: Optional[str] = None
    status: str = "uploading"
    analysis: Optional[BookAnalysis] = None

File: packages/shared/schemas/project.ts
export interface Character {
  name: string;
  description: string;
  visual_description: string;
}

export interface Scene {
  title: string;
  description: string;
  quote?: string;
  mood: string;
  visual_description: string;
  scene_type: 'introduction' | 'character_reveal' | 'tension_build' | 'conflict' | 'climax' | 'emotional_pause' | 'ending_hook';
}

export interface BookAnalysis {
  summary: string;
  themes: string[];
  genre: string;
  mood: string;
  target_audience: string;
  characters: Character[];
  key_scenes: Scene[];
}

export interface Project {
  id: string;
  title: string;
  author: string;
  description: string;
  book_file_url?: string;
  cover_image_url?: string;
  status: 'uploading' | 'analyzing' | 'planning' | 'editing' | 'rendering' | 'done';
  analysis?: BookAnalysis;
  created_at: string;
  updated_at: string;
}
